src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -39,74 +55,300 @@ class FilesRelationTest extends FlatSpec with Matchers with SparkTest {
+      } catch {
+        case NonFatal(_) => // ignore
+      }
+    }
+  }
+
+  it should "support upserts" taggedAs WriteTest in {
+    val source = s"upsert-${shortRandomString()}"
+
+    try {
+      //insert data
+      spark
+        .sql(s"""
+             |select "upsert-example" as name,
+             |'$source' as source,
+             |'externalId-$source' as externalId,
+             |null as id,
+             |null as mimeType
+     """.stripMargin)
+        .write
+        .format("cognite.spark.v1")
+        .option("apiKey", writeApiKey)
+        .option("type", "files")
+        .option("onconflict", "upsert")
+        .save()
+
+      val id = writeClient.files.retrieveByExternalId(s"externalId-$source").id
+
+      val insertWithUpsertIds =
+        retryWhile[Array[Row]](
+          spark
+            .sql(s"select * from destinationFiles where source = '$source'")
+            .collect,
+          df => df.length < 1)
+      assert(insertWithUpsertIds.length == 1)
+
+      //update data
+      spark
+        .sql(s"""
+             |select ${id.toString} as id,
+             |'text/plain-$source' as mimeType,
+             |'upserted-$source' as source
+     """.stripMargin)
+        .write
+        .format("cognite.spark.v1")
+        .option("apiKey", writeApiKey)
+        .option("type", "files")
+        .option("onconflict", "upsert")
+        .save()
+
+      //check updated data
+      val updatedWithUpsert =
+        retryWhile[Array[Row]](
+          spark
+            .sql(
+              s"select * from destinationFiles where mimeType = 'text/plain-$source' and source = 'upserted-$source'")
+            .collect,
+          df => df.length < 1)
+      assert(updatedWithUpsert.length == 1)
+
+      //original data doesn't exist
+      val updated = retryWhile[Array[Row]](
+        spark
+          .sql(s"select * from destinationFiles where source = '$source'")
+          .collect,
+        df => df.length > 0)
+      assert(updated.isEmpty)
+    } finally {
+      try {
+        cleanupFiles(s"upserted-$source")
+      } catch {
+        case NonFatal(_) => // ignore
+      }
+    }
 
   }
+
+  it should "support deletes" taggedAs WriteTest in {
+    val source = s"delete-${shortRandomString()}"
+
+    try {
+      spark
+        .sql(s"""
+                |select "name-$source" as name,
+                |null as id,
+                |'$source' as source,
+                |'externalId-$source' as externalId,
+                |null as mimeType,
+                |null as metadata,
+                |null as assetIds,
+                |null as datasetId,
+                |null as sourceCreatedTime,
+                |null as sourceModifiedTime,
+                |null as securityCategories,
+                |null as uploaded,
+                |null as createdTime,
+                |null as lastUpdatedTime,
+                |null as uploadedTime,
+                |null as uploadUrl
```

Comment:
Can just use `write...save()` instead, then we won't have to specify nulls like this.

Reply:
Here, I would be testing insertInto, will have write..save() everywhere else